features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/HttpPostMonitor.java

```diff
@@ -218,11 +218,11 @@ public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
                 if (Strresponse == null)
                     continue;
 
-                LOG.debug("HttpPostMonitor: banner = " + Strresponse);
-                LOG.debug("HttpPostMonitor: responseTime= " + responseTime + "ms");
+                LOG.debug("HttpPostMonitor: banner = {}", Strresponse);
+                LOG.debug("HttpPostMonitor: responseTime= {}ms", responseTime);
 
                 //Could it be a regex?
-                if (strBannerMatch.charAt(0)=='~'){
+                if (strBannerMatch != null && strBannerMatch.length() > 0 && strBannerMatch.charAt(0)=='~'){
```

Comment:
This can be simpliefied to `!Strings.isNullOrEmpty(strBannerMatch) && strBannerMatch.startsWith("~")`